_inc/lib/debugger/class-jetpack-cxn-test-base.php

```diff
@@ -151,23 +152,33 @@ public function run_tests() {
 	/**
 	 * Returns the full results array.
 	 *
-	 * @param string $group Testing group whose results we want. Defaults to "default" group. Use "all" for all tests.
+	 * @since 7.1.0
+	 * @since 7.3.0 Add 'type'
+	 *
+	 * @param string $type  Test type, async or direct.
+	 * @param string $group Testing group whose results we want. Defaults to all tests.
 	 * @return array Array of test results.
 	 */
-	public function raw_results( $group = 'default' ) {
+	public function raw_results( $type = 'all', $group = 'all' ) {
 		if ( ! $this->results ) {
 			$this->run_tests();
 		}
 
 		$results = $this->results;
 
-		if ( 'all' === $group ) {
-			return $results;
+		if ( 'all' !== $group ) {
+			foreach ( $results as $test => $result ) {
+				if ( ! in_array( $group, $result['group'], true ) ) {
+					unset( $results[ $test ] );
+				}
+			}
 		}
 
-		foreach ( $results as $test => $result ) {
-			if ( ! in_array( $group, $result['group'], true ) ) {
-				unset( $results[ $test ] );
+		if ( 'all' !== $type ) {
+			foreach ( $results as $test => $result ) {
+				if ( ! in_array( $type, $result['type'], true ) ) {
```

Comment:
This triggered an error for me:

```
in_array() expects parameter 2 to be array, string given
```

Reply:
550244fc9245b4c256a692e08db2d29a67ad3190